feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts

```diff
@@ -247,34 +247,34 @@ export abstract class ConfiguratorAttributeSingleSelectionBaseComponent extends
   ): string {
     let ariaLabel = '';
     if (value.valuePrice && value.valuePrice?.value !== 0) {
+      const translationKey = value.selected
+        ? 'configurator.a11y.selectedValueOfAttributeFullWithPrice'
+        : 'configurator.a11y.valueOfAttributeFullWithPrice';
       if (value.valuePriceTotal && value.valuePriceTotal?.value !== 0) {
         this.translation
-          .translate(
-            'configurator.a11y.selectedValueOfAttributeFullWithPrice',
-            {
-              value: value.valueDisplay,
-              attribute: attribute.label,
-              price: value.valuePriceTotal.formattedValue,
-            }
-          )
+          .translate(translationKey, {
+            value: value.valueDisplay,
+            attribute: attribute.label,
+            price: value.valuePriceTotal.formattedValue,
+          })
           .pipe(take(1))
           .subscribe((text) => (ariaLabel = text));
       } else {
         this.translation
-          .translate(
-            'configurator.a11y.selectedValueOfAttributeFullWithPrice',
-            {
-              value: value.valueDisplay,
-              attribute: attribute.label,
-              price: value.valuePrice.formattedValue,
-            }
-          )
+          .translate(translationKey, {
+            value: value.valueDisplay,
+            attribute: attribute.label,
+            price: value.valuePrice.formattedValue,
+          })
```

Comment:
@steinsebastian can you do me a favor to refactor the method,

- so there is only one call to the translation service. 
- that multi level if can be replacde by a simple if - else if - else if - else

Reply:
see also: https://github.com/SAP/spartacus/pull/18997/files#r1662749706